### ISSUE TITLE
[Automation] Method for keying off system state 

### ISSUE THREAD

**Issue by gkeishin** [openbmc/openbmc-test-automation#206]
Getting system states for code update and other poweron use cases
	- Need cleaner state methods to get the system state either using DBUS or REST

**Comment by sivassrr** [comment]
https://github.com/openbmc/openbmc/issues/727 - looks like dbus+ is getting introduced.

**Comment by gkeishin** [comment]
Brad is going to open up the discussion soon.

**Comment by gkeishin** [comment]
For BMC
https://github.com/openbmc/openbmc/issues/772

and for Host
https://github.com/openbmc/openbmc/issues/773

**Comment by sivassrr** [comment]
@rahulmah Could you update the estimation.

**Comment by rahulmah** [comment]
Here is the redme link https://github.com/openbmc/phosphor-dbus-interfaces/tree/master/xyz/openbmc_project/State for the state manager changes. 

I tested the patch with Chassis and Host changes of state manager and below are my observations.

1. Successfully able to power on HOST. At the same time able to see HOST in "Running" state, Chassis in "On" state
2. Successfully able to power off HOST. At the same time able to see HOST in "Off" state and Chassis in "Off" state
3. Successfully able to reboot HOST when it was in Running state. During reboot HOST went from Running->Off->Running state. I am also able to see obmcutil in HOST_BOOTED->HOST_POWERED_ON->HOST_BOOTING->HOST_BOOTED

Below are some observation which required your comments.
1. Unable to power on and off chassis (This is not working as you said) - When are we going to get this working? Are you dropping the changes any time soon.
2. Reboot HOST when it was in Off state. I was able to send Reboot to HOST. But here I am seeing that HOST state is changing state from Off to Running. Also state in obmcutil is changed from HOST_POWERED_OFF->HOST_POWERED_ON->HOST_BOOTING->HOST_BOOTED. I believe HOST should come back to same state after Reboot. So in this case, it should not power on after reboot.
3. We get RequestedHostTransition while checking host state. It does not make sense to display RequestedHostTransition here. I believe, the operation which we are triggering using RequestedHostTransition(like on, off, Reboot) should be more of a direct action and need not to be displayed. See the implementation for chassis in exiting code(/org/openbmc/control/chassis0).  Here we have action which can be performed on Chassis like powerOn,
